### PR TITLE
HPCC-17821 Zap report email option added to ECLWatch

### PR DESCRIPTION
**Pull request by kunalaswani**
Create email reporting option for ZAP Reports.

Signed-off-by: Aswani, Kunal <kunal.aswani@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

**Comment by kunalaswani** [comment]
@GordonSmith please review 
JIRA for backend response for sending email: https://track.hpccsystems.com/browse/HPCC-21547

**Comment by GordonSmith** [comment]
@kunalaswani I am not sure "The commit message title makes sense in a changelog, by itself" is true (might want to mention ZAP in there).

File: esp/src/eclwatch/WUDetailsWidget.js [hunk 1]
```diff
@@ -148,10 +155,21 @@ define([
...
+                this.emailCheckbox.set("value", "false");
+                this.emailSubject.reset();
+                this.emailBody.reset();
+            },
+
+            _onSubmitDialog: function(){
+                if (this.zapForm.validate()) {
+                    this.zapForm.set("action", "/WsWorkunits/WUCreateAndDownloadZAPInfo");
+                }
+                //this._onCancelDialog();
```
Comment (GordonSmith) [inline esp/src/eclwatch/WUDetailsWidget.js]: Remove commented code.

**Review by GordonSmith — CHANGES_REQUESTED** [review]
Code looks good - just one commented line to remove and I would suggest improving the commit title at the same time.

**Comment by kunalaswani** [comment]
@GordonSmith Changes made, please review. this.zapDialog.hide(); was added back into the _onsubmit area. thanks

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by kunalaswani** [comment]
@richardkchapman ok to merge

**Comment by richardkchapman** [comment]
We've been round this once before - and coded this option only to reject it. It's too big a security hole to allow arbitrary email targets (selected by the user) for zap reports.

**Pull request closed without merging**